src/microbe_stage/MicrobeSystem.cs

```diff
@@ -54,4 +57,63 @@ public void Process(float delta)
             microbe.ProcessSync(delta);
         }
     }
+
+    /// <summary>
+    ///   Tries to find specified Species as close to the point as possible.
+    /// </summary>
+    /// <param name="position">Position to search around</param>
+    /// <param name="species">What species to search for</param>
+    /// <param name="searchRadius">How wide to search around the point</param>
+    /// <returns>The nearest found point for the species or null</returns>
+    public (Microbe Microbe, Vector3 Position)? FindSpeciesNearPoint(Vector3 position, Species species,
+        float searchRadius = 200)
+    {
+        if (searchRadius < 1)
+            throw new ArgumentException("searchRadius must be >= 1");
+
+        (Microbe Microbe, Vector3 Position)? closestMicrobe = null;
+        float nearestDistanceSquared = float.MaxValue;
+        var searchRadiusSquared = searchRadius * searchRadius;
+        microbes ??= worldRoot.GetTree().GetNodesInGroup(Constants.RUNNABLE_MICROBE_GROUP).Cast<Microbe>()
+            .ToArray();
```

Comment:
```suggestion
        var searchRadiusSquared = searchRadius * searchRadius;

        microbes ??= worldRoot.GetTree().GetNodesInGroup(Constants.RUNNABLE_MICROBE_GROUP).Cast<Microbe>()
            .ToArray();
```

Just a bit extra blank space to separate logically separate variables, which looks better in my opinion. You don't have to accept this suggestion.